modules/core/src/test/java/org/apache/ignite/internal/processors/cache/persistence/snapshot/IgniteClusterSnapshotSelfTest.java

```diff
@@ -688,12 +694,11 @@ public void testRecoveryClusterSnapshotJvmHalted() throws Exception {
             grid2.context().cache().context().snapshotMgr().localSnapshotNames(null).isEmpty()
         );
 
-        ignite.snapshot().createSnapshot(SNAPSHOT_NAME)
-            .get();
+        createAndCheckSnapshot(ignite, SNAPSHOT_NAME);
 
         stopAllGrids();
 
-        IgniteEx snp = startGridsFromSnapshot(2, SNAPSHOT_NAME);
+        IgniteEx snp = startGridsFromSnapshot(onlyPrimary ? 3 : 2, SNAPSHOT_NAME);
```

Comment:
Why 3?

Reply:
Snapshot created in cluster of 3 nodes.
In `onlyPrimary` mode 2 node ins not enough to restore all keys.